docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
+      [description] [$amount] `. This will match when the optional flags are not included. An empty string will then be
+      returned.
+    - `-(in|out)\s+(.+)\$([\d.]+)\s+(.*)` checks for the corresponding structure `[-in/out] [whitepsace]
+      [description] [$amount] [remaining input]`. This will match when at least one of the optional flags are included,
+      and a string corresponding to `[remaining input]` will be returned.
+
+
+2. If an empty string is returned, an addCommand object will be returned from function addStatement. Otherwise,
+   a variable `String remainingInformation` will correspond to `[remaining input]` above. `remainingInformation` will
+   then
+   be checked for if it contains `-c` or `-date` flags.
+
+
+3. if `remainingInformation` contains a `-c` flag, it means that the user has provided a category for the item.
+   Thus, it will be passed into the function setCategory. setCategory will then use the following regex to match
+   `remainingInformation`:
+    - `-c\\s+(\\S+)` checks for the corresponding structure `[-c] [whitespace] [1-word category]`.
+    - `-c\\s+(\\S+)\\s+(.*)` checks for the same thing except category can be multiple words
+
+4. The last field to check is the `-date` field. If present, the setDate function will be called
+   on `remainingInformation`.
+   The setDate function will then use the following regex to match.
+    - `-date\\s+(\\d{2}/\\d{2}/\\d{4})` matches the corresponding
+      structure`[-date] [whitespace] [date in DD/MM/YYYY format]`
+
+These steps will ultimately parse the user's input and extract the necessary information. If any pattern does not match,
+our parser will throw a `RainDayException` indicating the wrong input format.
+
+#### Alternatives considered
+
+We have tried using the .split() function of strings to extract information. However, as we introduced more features and
+the commands grew more complex, the .split() function became very messy and inconvenient. Using this will also result in
+us needing to do more exceptions and error handling, which just made the entire process very complicated. Thus, we opted
+to use regular expressions, which is a more tidy and logical way to parse the inputs.
+
+### Viewing your data `view`
+
+- The command `view` is used to view all statements, and a ViewCommand object will be created.
+    - The format for the command is `view TIMESPAN -sort`
+    - Any other characters after any valid view command are automatically ignored
+- Information is presented in a table format to help improve clarity for users.
+    - The table includes information in the summary such as whether it is sorted and the amount of history shown. 
+    - This was deliberate as presenting this information at the bottom makes it easier for users to spot, 
+      as placing them at the top may cause users to miss it if they have a large table
+
+![ViewCommandSequenceDiagram.png](images\DeveloperGuide\ViewCommandSequenceDiagram.png)
```

Comment:
Perhaps for the method calls that does a `print` to the user, there is no need to have an arrow that points out to nothing. This is undefined syntax based on my understanding. I assume that these printing methods have a `void` signature. Might this be better if it was done as a self-invocation instead?

Sequence diagram in question:
![image](https://user-images.githubusercontent.com/69704055/228731329-39196856-e492-4f92-b9c5-5a56240b37b2.png)

Further, as mentioned earlier in the comment, it may be better to name these using your method names.

Reply:
Additionally, some of the self-invocation methods might be rendered wrongly.

For example:
![image](https://user-images.githubusercontent.com/69704055/228732280-00e553cb-d754-46bc-a052-22cb2a54ac91.png)


The callbacks do not use the activation bar syntax and are missing the return dotted error. Linked [here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/uml.html#self-invocation) is the syntax for self-invocation based on the textbook! 

Some of the diagrams in the DG do use the proper syntax. Might this be accidental?